test/torchaudio_unittest/prototype/io_test.py

```diff
@@ -0,0 +1,348 @@
+    get_wav_data,
+    save_wav,
+    skipIfNoFFmpeg,
+    nested_params,
+    is_ffmpeg_available,
+    get_image,
+    save_image,
+)
+
+if is_ffmpeg_available():
+    from torchaudio.prototype.io import (
+        Streamer,
+        SourceStream,
+        SourceVideoStream,
+        SourceAudioStream,
+    )
+
+
+def get_video_asset(file="nasa_13013.mp4"):
+    return get_asset_path(file)
+
+
+@skipIfNoFFmpeg
+class StreamerInterfaceTest(TempDirMixin, TorchaudioTestCase):
+    """Test suite for interface behaviors around Streamer"""
+
+    def test_streamer_invalid_input(self):
+        """Streamer constructor does not crash when receiving an invalid input"""
+        with self.assertRaises(RuntimeError):
+            Streamer("foobar")
+
+    def test_src_info(self):
+        """`get_src_stream_info` properly fetches information"""
+        s = Streamer(get_video_asset())
+        assert s.num_src_streams == 6
+
+        expected = [
+            SourceVideoStream(
+                media_type="video",
+                codec="h264",
+                codec_long_name="H.264 / AVC / MPEG-4 AVC / MPEG-4 part 10",
+                format="yuv420p",
+                bit_rate=71925,
+                width=320,
+                height=180,
+                frame_rate=25.0,
+            ),
+            SourceAudioStream(
+                media_type="audio",
+                codec="aac",
+                codec_long_name="AAC (Advanced Audio Coding)",
+                format="fltp",
+                bit_rate=72093,
+                sample_rate=8000.0,
+                num_channels=2,
+            ),
+            SourceStream(
+                media_type="subtitle",
+                codec="mov_text",
+                codec_long_name="MOV text",
+                format=None,
+                bit_rate=None,
+            ),
+            SourceVideoStream(
+                media_type="video",
+                codec="h264",
+                codec_long_name="H.264 / AVC / MPEG-4 AVC / MPEG-4 part 10",
+                format="yuv420p",
+                bit_rate=128783,
+                width=480,
+                height=270,
+                frame_rate=29.97002997002997,
+            ),
+            SourceAudioStream(
+                media_type="audio",
+                codec="aac",
+                codec_long_name="AAC (Advanced Audio Coding)",
+                format="fltp",
+                bit_rate=128837,
+                sample_rate=16000.0,
+                num_channels=2,
+            ),
+            SourceStream(
+                media_type="subtitle",
+                codec="mov_text",
+                codec_long_name="MOV text",
+                format=None,
+                bit_rate=None,
+            ),
+        ]
+        for i, exp in enumerate(expected):
+            assert exp == s.get_src_stream_info(i)
+
+    def test_src_info_invalid_index(self):
+        """`get_src_stream_info` should not crash when invalid values are passed"""
```

Comment:
should this be should crash or should not crash?

Reply:
I meant a crash in C++ layer, which is not recoverable from Python layer. Let me update the description.